File: src/backend/src/main/java/br/com/grupopipa/gestaointegrada/cadastro/usuario/UsuarioGridDTO.java
package br.com.grupopipa.gestaointegrada.cadastro.usuario;

import java.time.LocalDateTime;
import java.util.UUID;

import br.com.grupopipa.gestaointegrada.core.dto.GridDTO;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

@Builder
@Getter
@Setter
public class UsuarioGridDTO implements GridDTO {
    private UUID id;
    private String nome;
    private String login;
    private LocalDateTime createdAt;
}

File: src/frontend/src/app/components/cadastro/usuario/model/usuario-grid-dto.ts
export class UsuarioGridDTO {
  id: string;
  nome: string;
  login: string;
  createdAt: Date;
  deleted?: boolean;

  constructor(id: string, nome: string, login: string, createdAt: Date, deleted?: boolean) {
    this.id = id;
    this.nome = nome;
    this.login = login;
    this.createdAt = createdAt;
    this.deleted = deleted;
  }
}
